src/Runner.Worker/ExpressionFunctions/HashFiles.cs

```diff
@@ -0,0 +1,139 @@
+            if (context.State is DistributedTask.ObjectTemplating.TemplateContext templateContext &&
+                templateContext.ExpressionValues.TryGetValue(PipelineTemplateConstants.GitHub, out var githubContextData) &&
+                githubContextData is DictionaryContextData githubContext &&
+                githubContext.TryGetValue(PipelineTemplateConstants.Workspace, out var workspace) == true &&
+                workspace is StringContextData workspaceData)
+            {
+                string searchRoot = workspaceData.Value;
+                bool followSymlink = false;
+                string pattern = "";
+                if (Parameters.Count == 1)
+                {
+                    pattern = Parameters[0].Evaluate(context).ConvertToString();
+                }
+                else
+                {
+                    if (string.Equals(Parameters[0].Evaluate(context).ConvertToString(), "--followSymbolicLinks", StringComparison.OrdinalIgnoreCase))
+                    {
+                        followSymlink = true;
+                    }
+
+                    pattern = Parameters[1].Evaluate(context).ConvertToString();
+                }
+
+
+                // Convert slashes on Windows
+                if (s_isWindows)
+                {
+                    pattern = pattern.Replace('\\', '/');
+                }
+
+                context.Trace.Info($"Search root directory: '{searchRoot}'");
+                context.Trace.Info($"Search pattern: '{pattern}'");
+
+                string binDir = Path.GetDirectoryName(Assembly.GetEntryAssembly().Location);
+                string runnerRoot = new DirectoryInfo(binDir).Parent.FullName;
+
+                string node = Path.Combine(runnerRoot, "externals", "node12", "bin", "node");
+                if (s_isWindows)
+                {
+                    node = node + ".exe";
+                }
+                string hashFilesScript = Path.Combine(binDir, "hashFiles");
+                var hashResult = string.Empty;
+                var p = new ProcessInvoker(new FunctionTrace(context.Trace));
+                p.ErrorDataReceived += ((_, data) =>
+                {
+                    if (!string.IsNullOrEmpty(data.Data) && data.Data.StartsWith("__OUTPUT__") && data.Data.EndsWith("__OUTPUT__"))
+                    {
+                        hashResult = data.Data.Substring(10, data.Data.Length - 20);
+                        context.Trace.Info($"Hash result: '{hashResult}'");
+                    }
+                    else
+                    {
+                        context.Trace.Info(data.Data);
+                    }
+                });
+
+                p.OutputDataReceived += ((_, data) =>
+                {
+                    context.Trace.Info(data.Data);
+                });
+
+                var hashFilesArgs = $"\"{hashFilesScript.Replace("\"", "\\\"")}\"";
```

Comment:
have you considered env var so you dont have to deal with quoting

Reply:
I can use env to deal with glob pattern but still need to quote the entrypoint script.